app/controllers/api/v1x0/mixins/rbac_mixin.rb

```diff
@@ -54,11 +54,10 @@ def invalid_parameter(str)
           raise Catalog::InvalidParameter, str
         end
 
-        def catalog_admin_check
+        def role_check(role)
           return unless RBAC::Access.enabled?
 
-          access_obj = RBAC::Access.new("portfolios", "write").process
-          raise Catalog::NotAuthorized unless access_obj.catalog_admin?
+          raise Catalog::NotAuthorized unless RBAC::Roles.new("catalog").find(role)
```

Comment:
@lindgrenj6 This is not correct, it is using the roles used in Portfolio Sharing/Unsharing, instead you want a simple function that just has this

```
def role_present?(role_name)
    opts = { :limit => 100,
               :scope  => principal }
      RBAC::Service.call(RBACApiClient::RoleApi) do |api_instance|
        RBAC::Service.paginate(api_instance, :list_roles, opts).any? { |role| role.name == role_name }
      end
end 
```